docs/design/2018-10-24-view-support.md

```diff
@@ -5,65 +5,123 @@
+    The view SQL SECURITY characteristic. The value is one of DEFINER or INVOKER.
+* CheckOptionType  ref https://dev.mysql.com/doc/refman/5.7/en/view-check-option.html  
+    The WITH CHECK OPTION clause can be given for an updatable view to prevent inserts to rows for which the WHERE clause in the select_statement is not true. It also prevents updates to rows for which the WHERE clause is true but the update would cause it to be not true (in other words, it prevents visible rows from being updated to nonvisible rows).  
+    In a WITH CHECK OPTION clause for an updatable view, the LOCAL and CASCADED keywords determine the scope of check testing when the view is defined in terms of another view. When neither keyword is given, the default is CASCADED.
+* IsUpdatable  ref https://dev.mysql.com/doc/refman/5.7/en/view-updatability.html  
+    This parameter mark if this view is updatable.
+* SelectStmt
+    This string is the select sql statement after sql rewriter.
+* Cols
+    This string array is the view's column alias names.
+* TableInfo.Columns
+    `TableInfo.Columns` only stores view's column origin names, if no alias name specific, it stores the same values as `ViewInfo.Cols`.
+
+We add `ViewInfo` struct point which named `View` to `TableInfo`. If `&View` is nil, then this tableinfo is a base table, else this tableinfo is a view. 
+ 
 Let me describe more details about the view DDL operation:
-1. Create a VIEW  
-  1.1 Parse the create view statement and build a logical plan for select cause part. If any grammar error occurs, return errors to parser.   
-  1.2 Examine view definer's privileges. Definer should own both `CREATE_VIEW` and base table's `SELECT` privileges.  
-  1.3 If `ViewFieldList` cause is empty, then we should generate view column name from `SelectStmt` cause.  
-  1.4 Replace any wildcard with column name in `SELECT` cause.  
-  1.5 Test `SELECT` cause with some rules to see if this view is updateable.  
-  1.6 Store the view definition into following tables
-     * `INFORMATION_SCHEMA.VIEW_COLUMN_USAGE` lists one row for each column in a view including the base table of the column where possible 
-     * `INFORMATION_SCHEMA.VIEW_TABLE_USAGE` lists one row for each table used in a view 
-     * `INFORMATION_SCHEMA.VIEWS` lists one row for each view
-      
+1. Create VIEW  
+   This proposal only support following grammar to create view:
+   ```
+    CREATE
+        [OR REPLACE]
+        [ALGORITHM = {UNDEFINED | MERGE | TEMPTABLE}]
+        [DEFINER = { user | CURRENT_USER }]
+        [SQL SECURITY { DEFINER | INVOKER }]
+        VIEW view_name [(column_list)]
+        AS select_statement
+        [WITH [CASCADED | LOCAL] CHECK OPTION]
+    ```
+    1. Parse the create view statement and build a logical plan for select cause part. If any grammar error occurs, return errors to parser.   
+    2. Examine view definer's privileges. Definer should own both `CREATE_VIEW` and base table's `SELECT` privileges.  
```

Comment:
We should also mention two things here:
1. TiDB only supports CRATE_PRIV which do not distinguish CREATE_VIEW and CREATE_TABLE now.
2. SELECT_PRIV is checked during buildLogicalPlan already.

Reply:
CREATE_VIEW priv support should be implement as well.